daemon/container.go

```diff
@@ -30,7 +29,7 @@ import (
 //  If none of these searches succeed, an error is returned
 func (daemon *Daemon) GetContainer(prefixOrName string) (*container.Container, error) {
 	if len(prefixOrName) == 0 {
-		return nil, errors.NewBadRequestError(fmt.Errorf("No container name or ID supplied"))
+		return nil, validationError{fmt.Errorf("No container name or ID supplied")}
```

Comment:
It might be cleaner to have a specific error type for this case instead of wrapping a message in something.